src/Symfony/Component/Serializer/Normalizer/DenormalizerInterface.php

```diff
@@ -27,12 +27,14 @@ interface DenormalizerInterface
     /**
      * Denormalizes data back into an object of the given class.
      *
-     * @param mixed  $data    Data to restore
-     * @param string $type    The expected class to instantiate
-     * @param string $format  Format the given data was extracted from
-     * @param array  $context Options available to the denormalizer
+     * @template TObject of object
      *
-     * @return mixed
+     * @param mixed                        $data    Data to restore
+     * @param string|class-string<TObject> $type    The expected class to instantiate
+     * @param string|null                  $format  Format the given data was extracted from
+     * @param array                        $context Options available to the denormalizer
+     *
+     * @return TObject|mixed
```

Comment:
from a type point of view, that's the same than `mixed`, so this does not bring any value